lib/compiler/passes/generate-bytecode.js

```diff
@@ -614,8 +638,9 @@ function generateBytecode(ast) {
               : "\"" + js.stringEscape(part) + "\""
           ).join(", ")
         + "]";
-      const regexpIndex = addConst(regexp);
-      const expectedIndex = addConst(
+      const match = node.match | 0;
```

Comment:
please don't write code that other people can't read, then repeat it dozens of places across the codebase

Reply:
Explained above.

Reply:
No, it wasn't.

Reply:
It turns out this is a substitute for `(if node.?match === 0)`